src/test/controllerengine_test.cpp

```diff
@@ -36,6 +36,16 @@ class ControllerEngineTest : public MixxxTest {
                                         QScriptValueList());
     }
 
+    void processEvents() {
+        // QCoreApplication::processEvents() only processes events that were
+        // queued when the method was called. Hence, events that were emitted
```

Comment:
"Hence, all subsequent events that are emitted while processing those queued events will not be processed and are enqueued for the next event processing cycle. Calling processEvents() twice ensures that at least all queued *and* the next round of emitted events are processed."